azure/scope/machinepoolmachine.go

```diff
@@ -293,18 +307,27 @@ func (s *MachinePoolMachineScope) UpdateStatus(ctx context.Context) error {
 			APIVersion: node.APIVersion,
 		}
 
-		s.AzureMachinePoolMachine.Status.Ready = noderefutil.IsNodeReady(node)
+		nodeReady := noderefutil.IsNodeReady(node)
+		s.AzureMachinePoolMachine.Status.Ready = nodeReady
+		if nodeReady {
+			conditions.MarkTrue(s.AzureMachinePoolMachine, clusterv1.MachineNodeHealthyCondition)
+		} else {
+			conditions.MarkFalse(s.AzureMachinePoolMachine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeProvisioningReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
@CecileRobertMichon @devigned @mboersma Could use some feedback on which condition false reason we should be using here. Here are the reasons available:
```
// WaitingForNodeRefReason (Severity=Info) documents a machine.spec.providerId is not assigned yet.
WaitingForNodeRefReason = "WaitingForNodeRef"

// NodeProvisioningReason (Severity=Info) documents machine in the process of provisioning a node.
// NB. provisioning --> NodeRef == "".
NodeProvisioningReason = "NodeProvisioning"

// NodeNotFoundReason (Severity=Error) documents a machine's node has previously been observed but is now gone.
// NB. provisioned --> NodeRef != "".
NodeNotFoundReason = "NodeNotFound"

// NodeConditionsFailedReason (Severity=Warning) documents a node is not in a healthy state due to the failed state of at least 1 Kubelet condition.
NodeConditionsFailedReason = "NodeConditionsFailed"
```

Reply:
I believe in this case it would be `NodeConditionsFailed`. `NodeProvisioningReason` would be above if the NodeRef is nil and we can't get the node yet.